blocks/page_list/templates/itinerary/AddWalkToListDialog.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+export default class AddWalkToListDialog extends React.Component {
+  constructor(...args) {
+    super(...args);
+  }
+
+  propTypes() {
+    return {
+      lists: React.PropTypes.array,
+      add: React.PropTypes.func,
+      createList: React.PropTypes.func,
+      walkSelected: React.PropTypes.number,
+    }
+  }
+
+  render() {
+    const {lists, add, createList, activeWalk} = this.props;
+    //selectedWalk comes from where
+
+    const allLists = lists.map(list => {
+
+      const {id,title,walks} = list;
+
+      const walkFound = walks.find(walk => walk.id === activeWalk);
+
+      if(walkFound){
+        return ( //FIX: Issue with onClick, only available for last item in the list
+          <li onClick={(ev) => add(activeWalk,list,ev.target.value)} key={id}>
+            <span className="glyphicon glyphicon-ok">{title}</span>
+          </li>
+        )
+      } else {
+        return (
+          <li key={id}>
+            <span>{title}</span>
+          </li>
+        )
+      }
+    });
+
+    return(
+      <dialog id="addWalk">
+        <ul>
+          {allLists}
+        </ul>
+
+        <input ref="list" placeholder="Create a new List"></input>
```

Comment:
I'll just mention it once, but apparently using a string value for `ref` is  deprecated at this point, in favour of using a callback that sets the ref to a variable when initially mounted.

Using a blank input and referencing it from the DOM is also very bad practice - instead, give it an `onChange` handler and update its state, e.g. `<input value={this.state.new} onChange={ev => this.setState({new: ev.target.value})} />`